src/services/language-services/hover/widget-hover.ts

```diff
@@ -0,0 +1,464 @@
+import * as vscode from 'vscode';
+
+export const ACTIVITY_INDICATOR = vscode.languages.registerHoverProvider('html', {
```

Comment:
same comment here as for the other two hover implementations - there's a lot of duplicate code, which can be avoided.